src/Microsoft.Framework.Caching.Abstractions/MemoryCache/CacheEntryOptions.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.Caching.Memory
+{
+    public class CacheEntryOptions
+    {
+        private DateTimeOffset? _absoluteExpiration;
+        private TimeSpan? _absoluteExpirationRelativeToNow;
+        private TimeSpan? _slidingExpiration;
+
+        /// <summary>
+        /// Gets or sets an absolute expiration date for the cache entry.
+        /// </summary>
+        public DateTimeOffset? AbsoluteExpiration
+        {
+            get
+            {
+                return _absoluteExpiration;
+            }
+            set
+            {
+                if (AbsoluteExpirationRelativeToNow != null)
+                {
+                    throw new InvalidOperationException(
```

Comment:
Is this going to cause a conflict with AddEntryLink?  I don't think there's a problem with setting both at this stage, it's up to MemoryCache to resolve which one to use.